src/kagglehub/gcs_upload.py

```diff
@@ -136,32 +142,61 @@ def _upload_blob(file_path: str, model_type: str) -> str:
     return response["token"]
 
 
+def zip_files(source_path_obj: Path, zip_path: Path, update_queue: Queue) -> None:
+    def zip_file(file_path: Path) -> None:
+        arcname = file_path.relative_to(source_path_obj)
+        size = file_path.stat().st_size
+        with zipfile.ZipFile(zip_path, "a", zipfile.ZIP_STORED) as zipf:
+            zipf.write(file_path, arcname)
+        update_queue.put(size)
+
+    files = [file for file in source_path_obj.rglob("*") if file.is_file()]
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
no this zips into a single archive. because we open the zip file with the append, "a", flag

I compared this method to other methods and it seems be the best 
(fastest) temporary solution

what do you mean by benchmarking in this context?

Reply:
this logic essentialy does the following:
launch different threads for different files
each thread zips a file into the one zip file
and there is another thread that manages the queue of uploaded files to update the progress bar. else the progress bar will go from 0 to 100 and not show progress

Reply:
Just write a 1 pager with a table and plot of how long it takes while you vary: number of files, size of files, workers.

Reply:
https://docs.google.com/document/d/1YG6Z-Kg3PSv-u6EB5N5K1gRe6Cj9eDYOYSmfd0aQ1Ao/edit?tab=t.0